src/NullLogger.js

```diff
@@ -0,0 +1,32 @@
+/*
+ * NullLogger.js - Default logger that does not output any strings
+ *
+ * Copyright © 2023 JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @class null logger with a similar API as log4js which produces no output
+ */
+class NullLogger {
```

Comment:
I don't think NullLogger specifically should be part of this package.

Since logger is being provided by means of constructor dependency injection, the `Logger` interface becomes part of the overall API. Still, providing an actual *instance* of `Logger` is a responsibility of the app that wants to instantiate the dependency client (e.g. `Parser`).

To match the convention we already have (and because JS does not have interfaces), `NullLogger` could instead become an abstract class
```js
/** @abstract */
class Logger {
    /** @abstract */
    fatal(/** @type {string} */ message) {}
    
    // etc
}
```

`NullLogger` could then be a class only used for testing purposes
```js
class NullLogger extends Logger {
    /** @override */
    fatal(message) { /* no-op */ }
    // etc
}
```

while `ilib-lint` could plug in an own `Logger` instance which adapts log4js.

Reply:
removed